src/hooks/useAddress.ts

```diff
@@ -1,16 +1,10 @@
 import { useSafeAppsSDK } from '@gnosis.pm/safe-apps-react-sdk'
 
-import { useIsSafeApp } from '@/hooks/useIsSafeApp'
 import { useWallet } from '@/hooks/useWallet'
 
 export const useAddress = (): string | undefined => {
-  const isSafeApp = useIsSafeApp()
   const { safe } = useSafeAppsSDK()
   const wallet = useWallet()
 
-  if (isSafeApp) {
-    return safe.safeAddress
-  }
-
-  return wallet?.address
+  return safe.safeAddress || wallet?.address
```

Comment:
The initial delegation step was not loading the state correctly without this change.

`safe.safeAddress` is an empty string when using this as a DApp, meaning it uses `wallet?.address`.